lib/Predicate.js

```diff
@@ -101,6 +101,30 @@ function termMatches(text, terms, options) {
   return termMatchesTokens(text, terms, caseSensitive);
 }
 
+RegExp.escape = function(text) {
+  return text.replace(/[-[\]{}()*+?.,\\^$|#\s]/g, "\\$&");
+};
+
+function itemMatches(text, item, options) {
+  /**
+   * Check whether the item is in the text
+   *
+   * @param {string} text Input text
+   * @param {string} item Item to match
+   * @param {object} options Item analysis options
+   */
+  var separator = options.separator, regexp = options.separator_regexp,
+    items = [];
+  if (typeof regexp === 'undefined' || regexp == null) {
+    if (typeof separator === 'undefined' || separator == null) {
+      separator = " ";
+    }
+    regexp = RegExp.escape(separator);
```

Comment:
if you move the definition of RegExp.escape elsewhere, the duplicated logic here could also be refactored